src/components/contactsListView/ContactListItem.jsx

```diff
@@ -0,0 +1,39 @@
+import { useNavigate } from "react-router-dom";
+// import ".."
+
+export const ContactListItem = ({ contact, contacts, setContacts }) => {
+  const navigate = useNavigate();
+  const handleView = () => {
+    navigate(`/contacts/${contact.id}`);
+  };
+  const handleEdit = () => {
+    navigate(`/contacts/${contact.id}/edit`, { state: contact });
```

Comment:
better not to pass state and just re-fetch the contact in the edit page

Reply:
imagine being sent the url over discord, if you use state and I open the link you sent me, the state will be null because I open the page from discord